opentreemap/treemap/templatetags/species_thumbprint.py

```diff
@@ -0,0 +1,23 @@
+from django import template
+
+register = template.Library()
+
+last_request_and_thumbprint = (None, None)
+
+# Some pages contain many species autocompletes. To avoid re-querying
+# we hang on to the last request and thumbprint. This isn't thread-safe,
+# but the the worst that could happen with a data race is an extra query.
```

Comment:
I _think_ your concerns about thread-safety are unlikely, because AFAIK request objects are never shared between threads.  Like you say though, a data race here isn't so bad.